src/components/CropSelector/CropCalendarView/RenderCrops.js

```diff
@@ -0,0 +1,201 @@
+import React, { useContext } from 'react';
+import {
+  CropImage,
+  flipCoverCropName,
+  LightButton,
+  trimString,
+  getRating,
+} from '../../../shared/constants';
+import CropSelectorCalendarView from '../CropSelectorCalendarView';
+import { Context } from '../../../store/Store';
+import { Button, TableCell, TableRow } from '@mui/material';
+import '../../../styles/cropCalendarViewComponent.scss';
+
+const RenderCrops = ({ cropData, active, setModalOpen, modalOpen, setModalData }) => {
+  const { state, dispatch } = useContext(Context);
+
+  const selectedBtns = state.selectedCrops.map((crop) => {
+    return crop.id;
+  });
+
+  const hasGoalRatingTwoOrLess = (crop = []) => {
+    const { selectedGoals } = state;
+
+    return crop.inactive || selectedGoals.every((rating) => crop.fields[rating] <= 2);
+  };
+
+  const getAverageGoalRating = (selectedGoals, crop) => {
+    // get goal rating for each crop and calculate+render rating
+    let goalRating = 0;
+    selectedGoals.forEach((goal) => {
+      if (crop.fields[goal]) {
+        goalRating += crop.fields[goal];
+      }
+    });
+
+    return getRating(goalRating / selectedGoals.length);
+  };
+
+  const addCropToBasket = (cropId, cropName, btnId, cropData) => {
```

Comment:
can we remove the fact that the return is in a function returning null as else.  we don't need to define that.  and this wouldn't be the right way to do it.  (I know this isn't logic you created) lets see if this can be refactored to have the proper component structure.